include/libtorrent/block_cache.hpp

```diff
@@ -235,21 +235,14 @@ namespace aux {
 		// bitfield
 		std::uint16_t blocks_in_piece = 0;
 
-		// this is the number of threads that are currently holding
-		// a reference to this piece. A piece may not be removed from
-		// the cache while this is > 0
-		std::uint8_t piece_refcount = 0;
-
-		// 8 bytes padding
+		// the number of blocks in the cache for this piece
+		std::uint16_t num_blocks = 0;
 
 		// ---- 64 bit boundary ----
 
 		// the number of dirty blocks in this piece
 		std::uint32_t num_dirty:14;
 
-		// the number of blocks in the cache for this piece
-		std::uint32_t num_blocks:14;
```

Comment:
by bumping this to 16 bits, the max piece size goes to 512 MiB

Reply:
How about `num_dirty`?

Reply:
yeah, that should be wider too

Reply:
> How about `num_dirty`?

And `pinned`?

Reply:
> > How about `num_dirty`?
> 
> And `pinned`?

I hope you haven't lost sight of it. It is also 15 bits.

Reply:
I'm tempted to just fix the check rather than mucking about with this struct. Do you think there's a legitimate use for torrents with piece sizes this large? It really seems questionable

Reply:
> Do you think there's a legitimate use for torrents with piece sizes this large? It really seems questionable

I only know that the problem was reported by users... In addition, what seemed meaningless yesterday or today will become commonplace tomorrow.
Support for 512 MiB piece size does not require major changes, so why not add it?
Still, I don't insist. The correct processing of limits is good in itself. But I would still expand the potential capacity in upcoming major update.

Reply:
I think it would be much more reasonable to grow torrents in the number of pieces, rather than piece size.